version: '3'

services:
    jupyter-test:
      build:
          context: .
      command: ["jupyter", "notebook", "--ip=0.0.0.0", "--port=8888", "--allow-root"]
      ports:
        - 8888:8888
        - 8889:8889
      networks:
      - my-proxy-net
      volumes:
      - hdfs_hadoop:/hadoop-2.7.6
      - spark_spark:/spark-2.4.0-bin-hadoop2.7
      - ./:/notebooks
      environment:
      - SPARK_HOME=/spark-2.4.0-bin-hadoop2.7
      - HADOOP_CONF_DIR=/hadoop-2.7.6/etc/hadoop
      - PYSPARK_DRIVER_PYTHON=/usr/local/bin/python3.7
      - PYSPARK_PYTHON=/usr/local/bin/python3.7
      - JUPYTER_PATH=/notebooks
      

networks:
  my-proxy-net:
    external:
      name: custom_network
volumes:
    hdfs_hadoop:
        external: true
    spark_spark:
        external: true